src-ts/proposalStage.ts

```diff
@@ -594,6 +596,50 @@ export class L2TimelockExecutionBatchStage extends L2TimelockExecutionStage {
           arbOneSignerOrProvider
         );
         proposalStages.push(executeBatch);
+      } else if (log.topics[0] === timelockInterface.getEventTopic("CallScheduled")){
+        try {
+          const callScheduledArgs = timelockInterface.parseLog(log)
+            .args as CallScheduledEvent["args"];
+
+          const { data } = ArbSys__factory.createInterface().decodeFunctionData(
+            "sendTxToL1",
+            callScheduledArgs.data
+          );
+
+          const { salt } = this.decodeScheduleBatch(data);
+
+          // calculate the id and check if that operation exists
+          const operationId = this.hashOperationBatch(
```

Comment:
We're assuming here that there's only one operation in the batch, if that's not the case then id check will fail i think.